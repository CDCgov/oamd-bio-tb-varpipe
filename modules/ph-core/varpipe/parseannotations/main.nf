process VARPIPE_PARSEANNOTATIONS {

    tag "${meta.id}"
    label 'process_single'

    container 'oamd-bio-biocontainers-python:3.12_80c558c_v3'

    input:
    tuple val(meta), path(annotation)
    path(mutationloci)

    output:
    tuple val(meta), path("*_DR_loci_Final_annotation.txt"), optional:true, emit: drLociFinalAnnotation
    tuple val(meta), path("*_full_Final_annotation.txt")   , optional:true, emit: fullFinalAnnotation
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'parse_annotation.py'

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_DR_loci_Final_annotation.txt
    touch ${prefix}_full_Final_annotation.txt
    touch ${prefix}_DR_loci_annotation.txt
    touch ${prefix}_full_annotation.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
