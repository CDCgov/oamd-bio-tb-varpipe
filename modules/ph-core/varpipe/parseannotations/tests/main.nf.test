nextflow_process {

    name "Test Process VARPIPE_PARSEANNOTATIONS"
    script "../main.nf"
    process "VARPIPE_PARSEANNOTATIONS"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/parseannotations"

    test("varpipe parse DR loci annotation [vcf]") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_DR_loci_annotation.txt'),
                ]

                input[1] = [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/mutation_loci.txt') ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe parse full annotation [vcf]") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_full_annotation.txt')
                ]

                input[1] = [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/mutation_loci.txt') ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe parse full annotation [vcf] -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_full_annotation.txt')
                ]

                input[1] = [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/mutation_loci.txt') ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe parse DR loci annotation [vcf] -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_DR_loci_annotation.txt'),
                ]

                input[1] = [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/mutation_loci.txt') ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}

