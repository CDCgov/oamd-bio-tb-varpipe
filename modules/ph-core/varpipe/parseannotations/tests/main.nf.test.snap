{
    "varpipe parse full annotation [vcf]": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_Final_annotation.txt:md5,887915ba6876c0efe984bea7f9c60943"
                    ]
                ],
                "2": [
                    "versions.yml:md5,78a31018c2e8b194d02fc67f005b85ba"
                ],
                "drLociFinalAnnotation": [
                    
                ],
                "fullFinalAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_Final_annotation.txt:md5,887915ba6876c0efe984bea7f9c60943"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,78a31018c2e8b194d02fc67f005b85ba"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T15:41:19.777329043"
    },
    "varpipe parse DR loci annotation [vcf] -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_Final_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_Final_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,bf5f2704552773f2566947064409aa8c"
                ],
                "drLociFinalAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_Final_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fullFinalAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_Final_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bf5f2704552773f2566947064409aa8c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-05T11:11:24.605441105"
    },
    "varpipe parse DR loci annotation [vcf]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_Final_annotation.txt:md5,a51de6d8c32e239f3d0fe838801239d3"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,78a31018c2e8b194d02fc67f005b85ba"
                ],
                "drLociFinalAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_Final_annotation.txt:md5,a51de6d8c32e239f3d0fe838801239d3"
                    ]
                ],
                "fullFinalAnnotation": [
                    
                ],
                "versions": [
                    "versions.yml:md5,78a31018c2e8b194d02fc67f005b85ba"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T15:41:13.509965662"
    },
    "varpipe parse full annotation [vcf] -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_Final_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_Final_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,bf5f2704552773f2566947064409aa8c"
                ],
                "drLociFinalAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_Final_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fullFinalAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_Final_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bf5f2704552773f2566947064409aa8c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-05T11:11:19.839466875"
    }
}
